lib/macho/headers.rb

```diff
@@ -61,17 +69,57 @@ module MachO
   # @see http://llvm.org/docs/doxygen/html/Support_2MachO_8h_source.html
   CPU_SUBTYPE_LIB64 = 0x80000000
 
-  # all x86-type CPUs
+  # all CPU sub-types for CPU type `CPU_TYPE_I386`
   CPU_SUBTYPE_X86_ALL = 3
 
-  # all x86-type CPUs (what makes this different from CPU_SUBTYPE_X86_ALL?)
-  CPU_SUBTYPE_X86_ARCH1 = 4
+  # all CPU sub-types for CPU type `CPU_TYPE_X86_64`
+  CPU_SUBTYPE_X86_64_ALL = CPU_SUBTYPE_X86_ALL
+
+  # all CPU sub-types for CPU type `CPU_TYPE_X86_64` (Haskell tree)
+  CPU_SUBTYPE_X86_64_H = 8
+
+  # all CPU sub-types for CPU type `CPU_TYPE_ARM`
+  CPU_SUBTYPE_ARM_V7 = 9
+  CPU_SUBTYPE_ARM_V7S = 11
+  CPU_SUBTYPE_ARM_V7K = 12
+
+  # all CPU sub-types for CPU-type `CPU_TYPE_ARM64`
+  CPU_SUBTYPE_ARM64_ALL = 0
+
+  # PowerPC G3
+  CPU_SUBTYPE_POWERPC_750 = 9
 
-  # association of cpu subtypes to string representations
+  # PowerPC G4
+  CPU_SUBTYPE_POWERPC_7400 = 10
+
+  # PowerPC G4 "Voyager"
+  CPU_SUBTYPE_POWERPC_7450 = 11
+
+  # PowerPC G5
+  CPU_SUBTYPE_POWERPC_970 = 100
+
+  # all CPU sub-types for CPU type `CPU_TYPE_POWERPC`
+  CPU_SUBTYPE_POWERPC_ALL = 0
+
+  # all CPU sub-types for CPU type `CPU_TYPE_POWERPC64`
+  CPU_SUBTYPE_POWERPC64_ALL = CPU_SUBTYPE_POWERPC_ALL
+
+  # association of CPU types/subtype pairs to string representations
   CPU_SUBTYPES = {
-    CPU_SUBTYPE_X86_ALL => "CPU_SUBTYPE_X86_ALL",
-    CPU_SUBTYPE_X86_ARCH1 => "CPU_SUBTYPE_X86_ARCH1"
-  }
+    [CPU_TYPE_I386, CPU_SUBTYPE_X86_ALL] => "CPU_SUBTYPE_X86_ALL",
+    [CPU_TYPE_X86_64, CPU_SUBTYPE_X86_64_ALL] => "CPU_SUBTYPE_X86_64_ALL",
+    [CPU_TYPE_X86_64, CPU_SUBTYPE_X86_64_H] => "CPU_SUBTYPE_X86_64_H",
+    [CPU_TYPE_ARM, CPU_SUBTYPE_ARM_V7] => "CPU_SUBTYPE_ARM_V7",
+    [CPU_TYPE_ARM, CPU_SUBTYPE_ARM_V7S] => "CPU_SUBTYPE_ARM_V7S",
+    [CPU_TYPE_ARM, CPU_SUBTYPE_ARM_V7K] => "CPU_SUBTYPE_ARM_V7K",
+    [CPU_TYPE_ARM64, CPU_SUBTYPE_ARM64_ALL] => "CPU_SUBTYPE_ARM64_ALL",
+    [CPU_TYPE_POWERPC, CPU_SUBTYPE_POWERPC_750] => "CPU_SUBTYPE_POWERPC_750",
+    [CPU_TYPE_POWERPC, CPU_SUBTYPE_POWERPC_7400] => "CPU_SUBTYPE_POWERPC_7400",
+    [CPU_TYPE_POWERPC, CPU_SUBTYPE_POWERPC_7450] => "CPU_SUBTYPE_POWERPC_7450",
+    [CPU_TYPE_POWERPC, CPU_SUBTYPE_POWERPC_ALL] => "CPU_SUBTYPE_POWERPC_ALL",
+    [CPU_TYPE_POWERPC64, CPU_SUBTYPE_POWERPC_970] => "CPU_SUBTYPE_POWERPC_970",
+    [CPU_TYPE_POWERPC64, CPU_SUBTYPE_POWERPC64_ALL] => "CPU_SUBTYPE_POWERPC64_ALL"
```

Comment:
Since this covers the entirety of all CPU types and subtypes, I'm just going to get rid of get rid of `CPU_TYPES` and `CPU_SUBTYPES` in favor of `CPU_TYPE_TREE`.